force-app/main/default/classes/SOQL.cls

```diff
@@ -16,343 +16,463 @@
-	 * @param A set of Lead Ids.
-	 * @return A list of Accounts with matching email domains.
-	 */
-	public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
-		// Query the Leads
-
-		// Create a set of email domains from the Leads
-
-		// Loop through each Lead
-
-		// Get the email from the Lead
-
-		// Check if the email is not null
-
-		// Add the email domain to the set
-
-		// Query the Accounts
-		return null; // Replace null with the variable you used to store the result
-	}
-
-	/**
-	 * Question 23
-	 * Clones and Inserts Opportunity Line Items from an original Opportunity to a list of cloned Opportunities.
-	 * Adds a randomized number between 1 to 3 to the quantity of each line item.
-	 * Example: cloneLineItems('001XXXXXXXXXXXX', [list of opportunity ids]) should clone the line items from the Opportunity with the Id to each Opportunity in the list
-	 * @param Original Opportunity Id
-	 * @param List of cloned Opportunity Ids
-	 */
-	public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
-		// Query the original Opportunity Line Items
-
-		// Create a list to hold the new line items
-
-		// Generate a random number between 1 and 3
-
-		// Loop through each cloned Opportunity Id
-
-		// Loop through each original line item
-
-		// Clone the original line item and add the random number to the quantity
-
-		// Insert the new line items
-	}
-
-	/**
-	 * Question 24
-	 * Retrieves the Opportunity Name, Account Name, and Owner Name for Opportunities that have a Stage of 'Closed Won'.
-	 * @return A list of filtered Opportunity
-	 */
-	public static List<Opportunity> getOpportunityDetailsForClosedWon() {
-		return null; // Replace null with the variable you used to store the result
-	}
-
-	/**
-	 * Question 25
-	 * Sets the 'Primary_Contact__c' field for all Closed Won Opportunities to the Contact on the associated Account with the alphabetically first first name.
-	 * Example: if there are two Contacts on the Account, "Billy" and "Abby," the 'Primary_Contact__c' will be set to "Abby."
-	 */
-	public static void setPrimaryContact() {
-		//Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
-		// Create a set of Account Ids from the Opportunities
-
-		// Query the Contacts
-
-		// Loop through each Contact and add the Account Id and Contact to the map
-
-		// Loop through each Opportunity
-
-		// Update the Opportunity primary contact from accountToContactMap
-
-		// Update the Opportunities
-	}
+ public with sharing class SOQL {
+
+    /**
+     * Question 1
+     * Returns all leads from Salesforce Org
+     * @return List<Lead> List of all leads with Id, LastName, Company, Phone, Email
+     */
+    public static List<Lead> getAllLeads() {
+        try {
+            return [SELECT Id, LastName, Company, Phone, Email
+                    FROM Lead
+                    WHERE LastName != null
```

Comment:
Although not impossible because nothing is it's hard to find a lead in Salesforce without a last name.